### PR TITLE
Add map_2d

### PR DESCRIPTION
**Pull request by MartinKarp**
Adds the map_2d_t type and enables averaging fields into 2D as a post-processing step. Continues #1487 

Cleans up averaging in space contrib script a bit. Functionality is primarily tailored for the CPU, so don't expect any major performance from the averaging routines.

Next steps will be dependent on this functionality.

### CODE REVIEW

**Review by njansson — APPROVED** [review]
Should we have a (C) header in the files?

File: contrib/calc_lift_from_field/calc_lift_from_field.f90 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-!> Program to calculate the force and acting on a single boundary zone as well as
+	!> Program to calculate the force and acting on a single boundary zone as well as
```
Comment (timfelle) [inline contrib/calc_lift_from_field/calc_lift_from_field.f90:1]: Accidental tab it seems
```suggestion
!> Program to calculate the force and acting on a single boundary zone as well as
```

File: src/io/fld_file_data.f90 [hunk 2]
```diff
@@ -70,6 +72,71 @@ function fld_file_data_size(this) result(i)
...
+
+  end subroutine fld_file_init_same
+
+  !> Genereate same fields as in another fld_file
+  subroutine fld_file_init_n_fields(this, n_fields, n)
+    class(fld_file_data_t), target, intent(inout) :: this
+    integer, intent(in) :: n, n_fields
+    integer :: i, j
+    
+
```
Comment (timfelle) [inline src/io/fld_file_data.f90:111]: Whitespace
```suggestion


```

File: src/io/fld_file_data.f90 [hunk 3]
```diff
@@ -157,7 +224,9 @@ subroutine fld_file_data_free(this)
        do i = 1, this%n_scalars
           call this%s(i)%free()
        end do
+       deallocate(this%s)
     end if
+
```
Comment (timfelle) [inline src/io/fld_file_data.f90:229]: Whitespace
```suggestion

```

File: src/sem/map_1d.f90
Comment (timfelle) [inline src/sem/map_1d.f90:1]: There are quite a few lines that are rather long here.

File: src/sem/map_2d.f90
Comment (timfelle) [inline src/sem/map_2d.f90:1]: Also quite a lot of long lines here

**Review by timfelle — APPROVED** [review]
Generally seem pretty good. There are a few formatting things to consider as well as old print statements you probably didn't mean to include

**Comment by MartinKarp** [comment]
Thanks I will fix this!

**Pull request merged**